src/lib/serviceRoutines/postUpdateContext.cpp

```diff
@@ -541,8 +541,11 @@ std::string postUpdateContext
       }
       else
       {
-        ceP->contextAttributeVector[aIx]->stringValue = aP->stringValue; // FIXME P10: automacit value -> stringValue change, please review to check if it is safe
-        ceP->contextAttributeVector[aIx]->type  = aP->type;
+        ceP->contextAttributeVector[aIx]->stringValue    = aP->stringValue;
+        ceP->contextAttributeVector[aIx]->numberValue    = aP->numberValue;
+        ceP->contextAttributeVector[aIx]->boolValue      = aP->boolValue;
+        ceP->contextAttributeVector[aIx]->valueType      = aP->valueType;
+        ceP->contextAttributeVector[aIx]->compoundValueP = aP->compoundValueP == NULL ? NULL : aP->compoundValueP->clone();
```

Comment:
clone() means new memory being allocated... I'm not sure if and explicit release has to be performed or if it is already done as part of the release of the containing ContextAttribute object